src/frontend/citizen-portal/src/app/core/services/utils.service.ts

```diff
@@ -0,0 +1,124 @@
+import { DOCUMENT } from '@angular/common';
+import { Inject, Injectable } from '@angular/core';
+import { SortDirection } from '@angular/material/sort';
+
+import { WindowRefService } from './window-ref.service';
+
+export type SortWeight = -1 | 0 | 1;
+
+@Injectable({
+  providedIn: 'root',
+})
+export class UtilsService {
+  private window: Window;
+
+  constructor(
+    @Inject(DOCUMENT) private document: Document,
+    private windowRef: WindowRefService
+  ) {
+    this.window = this.windowRef.nativeWindow;
+  }
+
+  /**
+   * @description
+   * Scroll to the top of the mat-sidenav container.
+   */
+  public scrollTop() {
+    const contentContainer =
+      this.document.querySelector('.mat-sidenav-content') || this.window;
+    contentContainer.scroll({ top: 0, left: 0, behavior: 'smooth' });
+  }
+
+  /**
+   * @description
+   * Scroll to have the element in view.
+   */
+  public scrollTo(el: Element): void {
+    if (el) {
+      el.scrollIntoView({ behavior: 'smooth' });
+    }
+  }
+
+  /**
+   * @description
+   * Scroll to a material form field that is invalid, and if contained
+   * within a <section> scroll to the section instead.
+   */
+  public scrollToErrorSection(): void {
+    const firstElementWithError = document.querySelector(
+      'mat-form-field.ng-invalid'
+    );
+
+    if (firstElementWithError) {
+      const element =
+        firstElementWithError.closest('section') == null
```

Comment:
I typically try to avoid re-doing work. The `.closest('section')` is evaluated twice.

```javascript
if (firstElementWithError) {
  const section = firstElementWithError.closest('section');
  const element = section ?? firstElementWithError; // ?? was added in typescript 3.7
  this.scrollTo(element);
}